src/test/java/org/jabref/AutomaticRelinkTest.java

```diff
@@ -0,0 +1,174 @@
+        Files.createDirectory(B);
+
+        // Create A Directory & test file
+        A = main.resolve("A");
+        Files.createDirectory(A);
+        beforeFilePath = A.resolve("test.pdf");
+        Files.createFile(beforeFilePath);
+
+        MetaData metaData = new MetaData();
+        defaultDirectory = main.toAbsolutePath().toString();
+        metaData.setDefaultFileDirectory(defaultDirectory);
+        databaseContext = new BibDatabaseContext(new BibDatabase(), metaData);
+        Files.createFile(bibFolder.resolve("test.bib"));
+        databaseContext.setDatabasePath(bibFolder.resolve("test.bib"));
+
+        entry = new BibEntry();
+        LinkedFile linkedFile = new LinkedFile("desc", beforeFilePath, "");
+        List<LinkedFile> listLinked = new ArrayList<>();
+        listLinked.add(linkedFile);
+        entry.setFiles(listLinked);
+
+        filePreferences = mock(FilePreferences.class);
+        when(filePreferences.shouldStoreFilesRelativeToBibFile()).thenReturn(false); // Biblocation as Primary overwrites all other dirs, therefore we set it to false here
+    }
+
+    /**
+     * Run
+     * on
+     * empty
+     * Bib
+     * Entry
+     */
+    @Test
+    void checkEmpty() {
+        System.out.println(beforeFilePath);
+        BibEntry bib = new BibEntry();
+        AutomaticRelink.relink(bib, defaultDirectory);
+        BibEntry empty = new BibEntry();
+        assertEquals(empty, bib);
+    }
+
+    /**
+     * Check that relink returns the same entries
+     */
+    @Test
+    void checkSame() {
+        BibEntry bib = new BibEntry();
+        LinkedFile linkedFile = new LinkedFile("desc", beforeFilePath, "");
+        List<LinkedFile> listLinked = new ArrayList<>();
+        listLinked.add(linkedFile);
+        bib.setFiles(listLinked);
+        AutomaticRelink.relink(bib, defaultDirectory);
+        assertEquals(entry, bib);
+    }
+
+    /**
+     * Check that it stays the same once copied
+     */
+    @Test
+    void checkCopy() {
+        // Run "Automatically set file links" - check that the bib file was not modified
+        BibEntry bib = new BibEntry();
+        LinkedFile linkedFile = new LinkedFile("desc", beforeFilePath, "");
+        List<LinkedFile> listLinked = new ArrayList<>();
+        listLinked.add(linkedFile);
+        bib.setFiles(listLinked);
+        AutomaticRelink.relink(bib, defaultDirectory);
+        assertEquals(entry, bib);
+
+        // Copy Bib file from A to B
+        Path destination = B.resolve("test.pdf");
+        try {
+            Files.copy(A.resolve("test.pdf"), destination, StandardCopyOption.REPLACE_EXISTING);
+        } catch (
+                IOException e) {
+            e.printStackTrace();
+            fail("File copy failed!");
+        }
+
+        AutomaticRelink.relink(bib, defaultDirectory);
+        assertEquals(entry, bib);
+    }
+
+    /**
+     * Check that the file was relinked once moved
+     */
+    @Test
+    void relinkTest() {
```

Comment:
Avoid JavaDoc in test cases - the method names should be self-contained.

```suggestion
    @Test
    void fileIsRelinkedOnceMoved() {
```